### PR TITLE
feat: add the ability to disable ipv6 resolution for dns

### PR DESCRIPTION
**Pull request by LiquidTheDangerous**

### CODE REVIEW

**Comment by LiquidTheDangerous** [comment]
#239

File: util/args.go [hunk 1]
```diff
@@ -57,6 +58,7 @@ fragmentation for the first data packet and the rest
 		"pattern",
 		"bypass DPI only on packets matching this regex pattern; can be given multiple times",
 	)
+	flag.BoolVar(&args.DnsIpv6Enabled, "dns-ipv6-enabled", true, "enable DNS IPv6 resolution support")
```
Comment (LiquidTheDangerous) [inline util/args.go]: Or something like  `dns-ipv4-only`

Comment (xvzc) [reply util/args.go]: I think `dns-ipv4-only` would be better.

File: dns/dns.go [hunk 2]
```diff
@@ -72,6 +74,13 @@ func (d *Dns) ResolveHost(ctx context.Context, host string, enableDoh bool, useS
 	return "", fmt.Errorf("could not resolve %s using %s", host, clt)
 }
 
+func getQTypes(ipv6Enabled bool) []uint16 {
+	if ipv6Enabled {
+		return []uint16{dns.TypeAAAA, dns.TypeA}
+	}
+	return []uint16{dns.TypeA}
+}
+
```
Comment (xvzc) [inline dns/dns.go]: We can just remove this function and do this in `NewDns()`.

**Review by xvzc — APPROVED** [review]

**Pull request merged**